internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -243,22 +247,23 @@ func (e *EnvStackConfig) transformParameters(
 	// Remove or transform each of the current parameters.
 	var params []*cloudformation.Parameter
 	for k, p := range curr {
-		if transformed := transform(p, old[k]); transformed != nil {
-			params = append(params, transformed)
+		var currP = p
+		for _, transform := range transformFunc {
+			currP = transform(currP, old[k])
+		}
+		if currP != nil {
+			params = append(params, currP)
 		}
 	}
 	return params, nil
 }
 
-// transformEnvControllerParameters transforms a parameter such that it uses its previous value if:
-// 1. The parameter exists in the old template.
-// 2. The parameter is env-controller managed.
-// Otherwise, it returns the parameter untouched.
-func transformEnvControllerParameters(new cloudformation.Parameter, old *cloudformation.Parameter) *cloudformation.Parameter {
-	if old == nil { // The ParamKey doesn't exist in the old stack, use the new value.
-		return &new
+// transformEnvControllerParameters transforms an env-controller managed parameter.
+// If the parameter exists in the old template, it uses its previous value. Otherwise, it returns its new default value.
+func transformEnvControllerParameters(new *cloudformation.Parameter, old *cloudformation.Parameter) *cloudformation.Parameter {
+	if new == nil {
+		return nil
 	}
-
 	var (
 		isEnvControllerManaged = make(map[string]struct{})
 		exists                 = struct{}{}
```

Comment:
i know you didn't change this but 🥺 
```suggestion
```
and just change below to `isEnvControllerManaged[f] = struct{}{}`